server/main.go

```diff
@@ -48,32 +48,55 @@ func initLogger(loglevel int) {
 }
 
 /*
-	ARCHITECTURE
+	ARCHITECTURE OVERVIEW
 
 	Ultrablue is a client-server application, that operates over
 	Bluetooth Low Energy. This tool acts as the server.
-	Each step of the protocol is implemented in a characteristic,
-	and the client must read/write on those characteristic
-	successively to perform the remote attestation.
-	The protocol diagram can be found in the README.md file.
 
-	Characteristics implementation details:
+	Bluetooth Low Energy is a client-driven transport layer, in the sense that
+	the server exposes characteristics, and it is up to the client to read or
+	write them whenever it wants.
 
-		- Each characteristic is declared in its own file, ending with Chr.go.
+	Ultrablue does the opposite. The server only exposes one characteristic, and
+	internally keeps the current state of the protocol for a particular connection.
+	Whenever the client reads or write on that characteristic, the server will
+	choose what to send, and can refuse an operation, e.g. if a client tries to
+	read on the characteristic whereas a write operation was expected.
 
-		- As the chunking of the packets is not handled by the ble package,
-		each characteristic maintains its state in the context associated
-		with the connection.
-		It's up to the client to read/write enough times to complete the
-		operation, the server can't drive the communicatin and send every
-		chunks in a for loop.
+	When the server is run, the characteristic is exposed, and it starts accepting
+	clients.
+	When a client connects and performs it's first operation on the characteristic, 
+	a state is created for that client, and the protocol starts for it.
 
-		- The r/w handlers of the characteristics runs in a goroutine, thus
-		errors/success are transmitted to the main routine through channels.
+	---
+
+	KEY COMPONENTS
+
+	- The ultrablue characteristic: When a client perform a read/write operation
+	handlers the characteristic will process it through handlers. It implements
+	an abstraction for other parts of the server, which is a bidirectional go
+	channel.
+
+	- The state: On each characteristic interaction, the "state" will be fetched
```

Comment:
Start by saying what things are before saying how they behave:

every information that needs to be retained for the duration of a connection is stored in a State structure associated with a given client. The state is updated on each interaction with the characteristic. On the first ...